backend/core/src/main/java/org/sonarsource/sonarlint/core/ConnectionService.java

```diff
@@ -193,4 +195,10 @@ public List<SonarProjectDto> getAllProjects(Either<TransientSonarQubeConnectionD
       .collect(Collectors.toList());
   }
 
+  public String getProjectNameByKey(Either<TransientSonarQubeConnectionDto, TransientSonarCloudConnectionDto> transientConnection,
+    String projectKey, SonarLintCancelMonitor cancelMonitor) {
+    var serverApi = serverApiProvider.getForTransientConnection(transientConnection);
+    return serverApi.component().getProject(projectKey, cancelMonitor).map(ServerProject::getName).orElse("");
```

Comment:
Maybe returning null or an optional would better reflect that the project cannot be retrieved? If it was deleted for example.